services/lightbulb.c

```diff
@@ -0,0 +1,46 @@
+#include "jd_services.h"
+#include "interfaces/jd_pins.h"
+#include "jacdac/dist/c/lightbulb.h"
+
+struct srv_state {
+    SRV_COMMON;
+    uint16_t intensity;
+    uint8_t dimmable;
+    uint8_t pin;
+};
+
+REG_DEFINITION(                                   //
+    bulb_regs,                                   //
+    REG_SRV_COMMON,                         //
+    REG_U16(JD_LIGHT_BULB_REG_BRIGHTNESS),                         //
+    REG_OPT8(JD_LIGHT_BULB_REG_DIMMABLE),                         //  
```

Comment:
REG_OPT means that if the variable is set to `0` there will be no answer to queries to that register; this makes sense eg for variant register when the variant is unknown; however here, we always know if the light is dimmable or not, so it should not be optional

The register is marked as optional in the spec, so that some implementation may choose to skip it completely (actually, I don't thinks it's such a good idea for that one, but that's a separate discussion); if you've code to implement it and know the value, you should just always return it - so just use `REG_U8`.